src/binary/binaryFetcher/existingVersionHandler.ts

```diff
@@ -1,12 +1,20 @@
-import { promises as fs } from "fs";
+import { existsSync, promises as fs } from "fs";
+import * as path from "path";
 import sortBySemver from "../../utils/semver.utils";
 import { asyncFind } from "../../utils/utils";
 import isValidBinary from "./binaryValidator";
 import { getRootPath, versionPath } from "../paths";
+import { setDirectoryFilesAsExecutable } from "../utils";
 
 export default async function handleExistingVersion(): Promise<string | null> {
   try {
     const versionPaths = await fs.readdir(getRootPath());
+    await Promise.all(
+      versionPaths
+        .map((version) => path.dirname(versionPath(version)))
+        .filter((p) => existsSync(p))
+        .map(async (p2) => setDirectoryFilesAsExecutable(p2))
```

Comment:
change to `map(setDirectoryFilesAsExecutable)`

Reply:
must say it is a bit strange to manage it here and the downloading in a different part of the code, id suggest to to make the files  executable after  you bundle them, somewhere in the ci. and keep the  downloading/bundling/chmod together because they are related

Reply:
Not possible. Chmod is platform dependent and that is why the downloader does it as well. The chmod has to occur somewhere and this is the first place it is possible. I have to admit that the "duplicated" code was simpler by 10x and that is why i did it. the amount of changes required in order to not "duplicate" the path.join trivial code are immense in comparison  and not worth the "smell" of the dup. 
In other words IMO the cost of "duplicating" trivial code of 1 line vs changes that need to be propagated to tens of places are definitely not worth it

Reply:
1 -the downloader does the chmod bc the download happens after the installation.
2 - it is posible that we we did in the "bundled version, you can see it here https://github.com/codota/TabNine/blob/8ba545f20dac8de3b0fc0c6acf377a29e89bbcb4/dl_binaries.sh#LL23C30-L23C30

3 - disagree that this is a trivial code . it is is not bc you need to follow why it is here

Reply:
btw, the duplication was not in `path.join` it was in the calculation of the arch and the platform , which  just not makes any sense